### PR TITLE
feat: Add Event Timer module

### PR DESCRIPTION
**Pull request by ashishpuliyel**
## What does this change?
Adds an event timer module that can be imported and used globally.

## Why?
As part of the performance OKR we need additional monitoring of performance. 
Using this module gives us a single place to collate and handle events recorded for this

### CODE REVIEW

**Review by mxdvl — APPROVED** [review]
LGTM.

FYI: you don’t need the `!` in the PR title, unless you want to release a patch. `feat` will do just fine.

File: src/EventTimer.ts [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+	/**
+	 * Creates a new performance mark
+	 * For slot events also ensures each TYPE of event event is marked only once for 'first'
+	 * (the first time that event is triggered for ANY slot) and once for topAboveNav
+	 *
+	 * @param {string} eventName - The short name applied to the mark
+	 * @param {origin} [origin=page] - Either 'page' (default) or the name of the slot
+	 */
+	trigger(eventName: string, origin = 'page'): void {
+		const TRACKEDSLOTNAME = 'top-above-nav';
```
[the remark this answers is not in the document]
Comment (ashishpuliyel) [reply src/EventTimer.ts:78]: Can it be used on line 23?

**Review by ioanna0 — COMMENTED** [review]
would be great to have some unit tests on this, but i'm ok to move them on a subsequent PR

**Comment by ashishpuliyel** [comment]
Adding the event timer module

**Comment by ioanna0** [comment]
main failed on test coverage step: 
[COVERAGE DECREASED (-24.4%) TO 17.241%](https://coveralls.io/builds/36789529)

**Pull request merged**